airflow/models/baseoperator.py

```diff
@@ -636,51 +636,71 @@ def __setstate__(self, state):
         self.__dict__ = state
         self._log = logging.getLogger("airflow.task.operators")
 
-    def render_template_from_field(self, attr, content, context, jinja_env):
+    def render_template_fields(self, context: Dict) -> None:
         """
-        Renders a template from a field. If the field is a string, it will
-        simply render the string and return the result. If it is a collection or
-        nested set of collections, it will traverse the structure and render
-        all elements in it. If the field has another type, it will return it as it is.
+        Template all attributes listed in template_fields. Note this operation is irreversible.
+
+        :param context: Dict with values to apply on content
+        :type context: dict
+        """
+        if self.template_fields:
+            if self.dag.user_defined_macros:
+                context.update(self.dag.user_defined_macros)
```

Comment:
We shouldn't need to do this -- as `dag.get_template_env()` will handle this for us.

Reply:
Removed 👍